astropy/io/fits/tests/test_connect.py

```diff
@@ -273,8 +273,13 @@ def setup_class(self):
         hdu1 = PrimaryHDU()
         hdu2 = BinTableHDU(self.data1, name='first')
         hdu3 = BinTableHDU(self.data2, name='second')
+        hdu4 = ImageHDU(np.ones((3, 3)), name='third')
 
-        self.hdus = HDUList([hdu1, hdu2, hdu3])
+        self.hdus = HDUList([hdu1, hdu2, hdu3, hdu4])
+        self.hdub = HDUList([hdu1, hdu4, hdu3, hdu2])
+        self.hdu3 = HDUList([hdu1, hdu4, hdu3])
+        self.hdu2 = HDUList([hdu1, hdu2, hdu4])
+        self.hdu1 = HDUList([hdu1, hdu2])
```

Comment:
Using `self.hdu1` (singular, same for 2 and 3) is confusing. Could you change to `self.hdus1` (plural) ?